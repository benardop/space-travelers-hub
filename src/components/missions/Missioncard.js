import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { unjoinMission, joinMission } from '../../redux/missions/MissionsSlice';
import '../stylesheets/Missioncard.css';

const Missioncard = ({
  missionname, description, id, missionjoin,
}) => {
  const dispatch = useDispatch();

  const missionSwitchHandler = (e) => {
    dispatch(joinMission(e.target.id));
  };

  const unjoinswitchhandler = (e) => {
    dispatch(unjoinMission(e.target.id));
  };
  return (
    <div className="mission-card">
      <div className="mission">
        <div className="tbody">
          <div className="left">{missionname}</div>
          <div className="center">
            {description}
          </div>
          <div className="right">
            {missionjoin === false ? (<span className="badge2">Not A Member</span>) : (<span className="badge">Active Member</span>)}
            {missionjoin === false ? (<button className="join" type="submit" id={id} onClick={missionSwitchHandler}>Join Mision</button>) : (<button className="join-btn" type="submit" id={id} onClick={unjoinswitchhandler}>Leave Mission</button>)}
          </div>
        </div>
      </div>
    </div>
  );
};

Missioncard.propTypes = {
  id: PropTypes.string.isRequired,
  missionname: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  missionjoin: PropTypes.bool.isRequired,
};

export default Missioncard;
