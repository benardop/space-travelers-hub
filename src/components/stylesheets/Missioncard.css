.mission-card {
  width: 98vw;
  border: 0;
  margin: 0 auto;
  box-sizing: border-box;
}

.mission {
  margin: 0 auto;
}

.heading {
  width: 98vw;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 0 auto;
  margin-top: 10px;
  font-weight: 700;
}

.tbody {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.left {
  display: flex;
  flex: 0.15;
  justify-content: center;
  font-weight: 700;
  padding: 5px;
  border: 1px solid rgb(185, 184, 184);
}

.center {
  display: flex;
  flex: 1;
  align-items: center;
  border: 1px solid rgb(185, 184, 184);
  word-wrap: break-word;
  padding: 5px;
}

.right {
  display: flex;
  flex: 0.3;
  padding: 5px;
  align-items: center;
  justify-content: center;
  border: 1px solid rgb(185, 184, 184);
}

.badge {
  display: flex;
  margin: 5px;
  color: white;
  border-radius: 5px;
  background-color: rgb(109, 117, 125);
  font-size: 12px;
  height: 30px;
  align-items: center;
  justify-content: center;
}

.badge2 {
  display: flex;
  margin: 5px;
  color: white;
  border-radius: 5px;
  background-color: #40a2b8;
  font-size: 12px;
  height: 30px;
  align-items: center;
  justify-content: center;
}

.join-btn {
  font-size: 15px;
  padding: 5px;
  cursor: pointer;
  color: red;
  border: 1px solid red;
}

.join {
  font-size: 15px;
  padding: 5px;
  cursor: pointer;
}

@media screen and (max-width: 768px) {
  .heading {
    width: 98vw;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    margin: 0 auto;
    margin-top: 10px;
  }

  .left {
    display: flex;
    min-width: 20vw;
    padding: 5px;
    justify-content: center;
    border: 1px solid rgb(185, 184, 184);
    font-size: 0.75rem;
  }

  .center {
    display: flex;
    width: 60vw;
    align-items: center;
    border: 1px solid rgb(185, 184, 184);
    word-wrap: break-word;
    padding: 5px;
    font-size: 0.75rem;
  }

  .right {
    display: flex;
    width: 20vw;
    justify-content: center;
    flex-wrap: wrap;
    padding: 5px;
    border: 1px solid rgb(185, 184, 184);
    font-size: 0.75rem;
  }

  .join-btn {
    margin-top: 20px;
  }

  .badge2,
  .badge {
    text-align: center;
    padding: 5px;
  }
}
