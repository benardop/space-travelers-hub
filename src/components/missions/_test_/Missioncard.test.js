import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import store from '../../../redux/configureStore';
import Missioncard from '../Missioncard';

// test the mission card
const missioncard = [
  {
    missionid: '9D1B7E0',
    missionName: 'Thaicom',
    missionDesc:
    'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
    missionweblink: 'https://en.wikipedia.org/wiki/Thaicom',
    missionjoin: true,
  }];

const MockMissioncard = () => (
  <Provider store={store}>
    <Missioncard missioncard={missioncard} />
  </Provider>
);

test('renders data in the card as from API', () => {
  const tree = renderer.create(<MockMissioncard />).toJSON();
  expect(tree).toMatchSnapshot();
});

// check button change text
describe('when you click button it should change to leave mission', () => {
  it('should render missions page content', () => {
    render(
      (<MockMissioncard />),
    );
    userEvent.click(screen.getByRole('button', { name: /leave mission/i }));
    expect(screen.getByText(/active member/i)).toBeInTheDocument();
  });
});
