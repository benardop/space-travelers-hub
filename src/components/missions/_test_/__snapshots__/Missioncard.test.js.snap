// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders data in the card as from API 1`] = `
<div
  className="mission-card"
>
  <div
    className="mission"
  >
    <div
      className="tbody"
    >
      <div
        className="left"
      />
      <div
        className="center"
      />
      <div
        className="right"
      >
        <span
          className="badge"
        >
          Active Member
        </span>
        <button
          className="join-btn"
          onClick={[Function]}
          type="submit"
        >
          Leave Mission
        </button>
      </div>
    </div>
  </div>
</div>
`;
